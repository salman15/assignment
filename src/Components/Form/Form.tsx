import React from "react";
import styled, { css } from "styled-components";

interface FormProps {
  title?: string;
  description?: string;
  children: React.ReactNode;
  error?: boolean;
  submit?: Function;
}

const Form: React.FC<FormProps> = ({
  title,
  description,
  children,
  error = false,
  submit = () => {},
}) => {
  return (
    <FormContainer error={error} onSubmit={(e) => submit(e)}>
      {title && <h1>{title}</h1>}
      {description && <p>{description}</p>}
      {children}
    </FormContainer>
  );
};

export default Form;

const FormContainer = styled.form<{ error: boolean }>`
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  border-radius: 8px;
  background-color: white;
  padding: 8px;
  width: 100%;
  max-width: 450px;
  box-shadow: 0 2.8px 2.2px rgba(0, 0, 0, 0.034),
    0 6.7px 5.3px rgba(0, 0, 0, 0.048), 0 12.5px 10px rgba(0, 0, 0, 0.06),
    0 22.3px 17.9px rgba(0, 0, 0, 0.072), 0 41.8px 33.4px rgba(0, 0, 0, 0.086),
    0 100px 80px rgba(0, 0, 0, 0.12);
  ${(props) =>
    props.error &&
    css`
      border: solid 2px red;
    `};
`;
