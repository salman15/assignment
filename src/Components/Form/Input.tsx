import React from "react";
import styled from "styled-components";

interface InputProps {
  label?: string;
  warning?: boolean;
  warningText?: string;
  placeholder: string;
  inputValue?: string | number;
  setInputValue: Function;
  required?: boolean;
}

const Input: React.FC<InputProps> = ({
  label,
  placeholder,
  inputValue,
  setInputValue,
  warning,
  warningText,
  required,
}) => {
  return (
    <InputContainer>
      {label && (
        <Label>
          {label} {required && `*`}
        </Label>
      )}
      <InputLayout
        placeholder={placeholder}
        value={inputValue}
        onChange={(e: { target: { value: any } }) =>
          setInputValue(e.target.value)
        }
      />
      {warning && <Warning>{warningText}</Warning>}
    </InputContainer>
  );
};

export default Input;

const InputLayout = styled.input``;

const Label = styled.p`
  font-weight: bold;
  margin-bottom: 8px;
  font-size: 12px;
`;

const Warning = styled(Label)`
  color: red;
`;

const InputContainer = styled.div`
  display: flex;
  align-items: flex-start;
  justify-content: center;
  flex-direction: column;
`;
