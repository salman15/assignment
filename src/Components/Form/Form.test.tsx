import React from "react";
import { render, screen } from "@testing-library/react";
import Form from "./Form";

test("checks content of the form", () => {
  render(
    <Form>
      <p>I'm a form</p>
    </Form>
  );
  const linkElement = screen.getByText(/I'm a form/i);
  expect(linkElement).toBeInTheDocument();
});
