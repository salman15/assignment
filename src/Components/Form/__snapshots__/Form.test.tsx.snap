// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Form component Test if form component renders with children 1`] = `
<div>
  <form
    class="sc-bdfBwQ iHfWmQ"
  >
    <p>
      I'm a form
    </p>
    <div
      class="sc-jSgupP dehafT"
    >
      <input
        aria-label="text-input"
        autocomplete="text"
        class="sc-gsTCUz btDedy"
        name="label"
        placeholder="I'm a placeholder"
        type="text"
        value="I'm the input value"
      />
    </div>
  </form>
</div>
`;
